balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -548,7 +575,12 @@ func (b *pickfirstBalancer) updateSubConnState(sd *scData, newState balancer.Sub
 		return
 	}
 
+	if oldState == connectivity.Connecting && newState.ConnectivityState == connectivity.TransientFailure {
```

Comment:
I don't know, you and Easwar have much more context than me. All I know is Connecting -> TF means a connection attempt failed. Is there any harm in keeping the old state check?

Reply:
You can move a similar check for TF from above here and combine them.
```
// Record a connection attempt when exiting CONNECTING.
if newState.ConnectivityState == connectivity.TransientFailure {
	sd.connectionFailedInFirstPass = true
}
```

Reply:
I agree with you, but I think this might be a more accurate metrics it lies after the conditional check of "previously relevant subconns". So if you think this metric is semantically accurate above that check, I can move it, but I think this is more accurate metric (ignores "obsolete SubConns").

Reply:
Yes, I'm suggesting moving the following block down, instead of moving the new code up.
```go
if newState.ConnectivityState == connectivity.TransientFailure {
	sd.connectionFailedInFirstPass = true
}
```

Reply:
Ah, I see. Done.